### PR TITLE
feat(inventory): feature/icky 641 implemented nearby store with inventory for pdp

### PR DESCRIPTION
**Pull request by TriptiVermaXP**

### CODE REVIEW

File: components/dialogs/Store/SearchStore/SearchStore.tsx [hunk 1]
```diff
@@ -49,20 +48,42 @@ const SearchStore = (props: SearchStoreProps) => {
...
+          <Box>
+            <ProductItem
+              image={productGetters.getCoverImage(product as ProductCustom)}
+              name={productGetters.getName(product as ProductCustom)}
+              price={productGetters.getPrice(product as ProductCustom).regular?.toString()}
+              salePrice={productGetters.getPrice(product as ProductCustom).special?.toString()}
+              qty={quantity}
+            />
+          </Box>
+        )}
```
Comment (kibo-sushant) [inline components/dialogs/Store/SearchStore/SearchStore.tsx:86]: Storybook do not allow to type in Search
Without Inventory - should user able to open Store Info accordian?
                    Get Directions is not disabled

File: components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
 
 import type { Location, Maybe } from '@/lib/gql/types'
 
 interface StoreLocatorProps {
   isOpen: boolean
   isDialogCentered: boolean
+  title?: string
+  showProductAndInventory?: boolean
+  product?: ProductCustom
+  quantity?: number
```
Comment (kibo-sushant) [inline components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx:23]: No mandatory Prop?

File: __mocks__/stories/ProductLocationInventoryMock.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    items: [
+      {
+        productCode: 'BackP_006',
+        locationCode: 'RICHMOND',
+        stockAvailable: 100,
+        softStockAvailable: null,
+      },
+    ],
+  },
+}
```
Comment (kibo-sushant) [inline __mocks__/stories/ProductLocationInventoryMock.ts:16]: As per the conventions used for other mock files, file name should be LocationInventoryCollectionMock

File: components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx [hunk 4]
```diff
@@ -30,7 +44,15 @@ const StoreLocatorDialog = (props: StoreLocatorProps) => {
   const [searchParams, setSearchParams] = useState<{ filter: string }>({ filter: '' })
   const [searchTerm, setSearchTerm] = useState<string>('')
   const [selectedStore, setSelectedStore] = React.useState<LocationCustom>({})
+
   const { isError, data: locations } = useStoreLocations(searchParams)
+  const locationCodes: string = storeLocationGetters.getLocationCodes(
+    locations as Maybe<Location>[]
+  )
+  const locationInventory = useProductLocationInventory(
```
Comment (kibo-kevinwatts) [inline components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx:52]: destructure like we do elsewhere,
`const { data: locationInventory } = useProductLocationInventory(`

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 5]
```diff
@@ -66,6 +67,10 @@ const ProductDetailTemplate = (props: ProductDetailTemplateProps) => {
     product,
     purchaseLocation,
   })
+  const productInventory = useProductLocationInventory(
```
Comment (kibo-kevinwatts) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:70]: destructure like we do with other react query hooks

File: hooks/queries/useProductLocationInventory/useProductLocationInventory.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+export const useProductLocationInventory = (
+  productCode: string,
+  locationCodes: string
+): UseProductLocationInventoryType => {
+  const {
+    data = [],
+    isLoading,
+    isSuccess,
+  } = useQuery(inventoryKeys.all, () => loadProductLocationInventory(productCode, locationCodes), {
```
Comment (kibo-kevinwatts) [inline hooks/queries/useProductLocationInventory/useProductLocationInventory.ts:34]: I think we need to include productCode / locationCodes in the key.

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]
nice work!

File: components/common/KiboRadio/KiboRadio.tsx [hunk 7]
```diff
@@ -50,7 +51,12 @@ export const KiboRadio = (props: KiboRadioProps) => {
               sx={{ width: 'fit-content', alignItems: align, ...sx }}
               key={radio.value + index}
               value={radio.value}
-              control={<Radio inputProps={{ 'aria-label': radio.name }} />}
+              control={
+                <Radio
+                  inputProps={{ 'aria-label': radio.name }}
+                  {...(radio.disabled && { disabled: radio.disabled })}
+                />
+              }
```
[the remark this answers is not in the document]
Comment (TriptiVermaXP) [reply components/common/KiboRadio/KiboRadio.tsx:59]: Done

File: __mocks__/stories/ProductLocationInventoryMock.ts (hunk 3 above)
Comment (TriptiVermaXP) [reply __mocks__/stories/ProductLocationInventoryMock.ts]: done

File: hooks/queries/useProductLocationInventory/useProductLocationInventory.ts (hunk 6 above)
Comment (TriptiVermaXP) [reply hooks/queries/useProductLocationInventory/useProductLocationInventory.ts]: Done

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx (hunk 5 above)
Comment (TriptiVermaXP) [reply components/page-templates/ProductDetail/ProductDetailTemplate.tsx]: Done

File: components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx (hunk 4 above)
Comment (TriptiVermaXP) [reply components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx]: Done

File: components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx (hunk 2 above)
Comment (TriptiVermaXP) [reply components/dialogs/Store/StoreLocatorDialog/StoreLocatorDialog.tsx:23]: we need to show product and qty details on store locator dialog for PDP only, so created the optional parameter.

File: components/common/ProductItem/ProductItem.tsx [hunk 8]
```diff
@@ -119,7 +119,7 @@ const ProductItem = (props: ProductItemProps) => {
               </Box>
 
               <Collapse in={mdScreen ? true : expanded} timeout="auto" unmountOnExit>
-                <ProductOptionList options={options} />
+                {options && <ProductOptionList options={options} />}
```
[the remark this answers is not in the document]
Comment (TriptiVermaXP) [reply components/common/ProductItem/ProductItem.tsx:122]: Done

File: components/dialogs/Store/SearchStore/SearchStore.tsx (hunk 1 above)
Comment (TriptiVermaXP) [reply components/dialogs/Store/SearchStore/SearchStore.tsx:86]: SearchTerm is passing value from Parent comp (storeLocatorDialog) and maintained the state on parent itself, that's why search input box is not functioning into the storybook here. Should we have another state in this comp (middle level comp) for storybook? 
It's working in Parent component's (storeLocatorDialog) storybook. 

Purposely not disabled the Store info accordion for no inventory location. Please suggest if we should disable it.
Now grayed out the Get Directions text color.

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**